lib/openshift/report.rb

```diff
@@ -1,46 +1,69 @@
 # for reference on how to use Metering please consult the following link
 # https://github.com/operator-framework/operator-metering/blob/master/Documentation/using-metering.md
+
+require 'openshift/flakes/conditions'
+
 module BushSlicer
   class Report < ProjectResource
     RESOURCE = "reports"
 
-    def finished?(user: nil, cached: false, quiet: false)
-      phase(user:user, cached: cached, quiet: quiet) == :finished
+
+    def conditions(user: nil, cached: false, quiet: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet).dig('status', 'conditions')
+      rr.map { |cond| Conditions.new cond }
+    end
+
+    def running?(user: nil, cached: false, quiet: false)
+      # the reason 'Scheduled' is an initial state what we
+      conditions.any? { |c| c.type == 'Running' and c.reason != 'Scheduled' }
     end
 
     # wait until the stauts of the Report becomes 'Finished'
-    def wait_till_finished(user: nil, quiet: false)
+    def wait_till_finished(user: nil, cached: false, quiet: false)
+      seconds = 60 # for PVs it can take longer than 30s
+      success = wait_for(seconds) do
+        finished?(user:user, cached: cached, quiet: quiet)
```

Comment:
cached should be hardcoded to `false` here. Otherwise we can enter an infinite loop (at least until timeout is reached)

Reply:
same applies to wait_till_running